File: 프로그래머스/2/12936. 줄 서는 방법/줄 서는 방법.java
import java.util.*;

class Solution {
    
    public int[] solution(int n, long k) {
        
        ArrayList<Integer> people = new ArrayList<>();
        long permu = 1;
        
        for(int i=1; i<=n; i++){
            permu *= i;
            people.add(i);
        }
        
        int index = 0;
        int answer[] = new int[n];
        
        while(people.size() > 0){
            
            if(people.size() == 1){
                answer[index] = people.get(0);
            }
            
            permu /= people.size();
            int num = (int)((k-1)/permu);
            answer[index] = people.remove(num);
            index++;
            k -= num*permu;
        }
        
        return answer;
        
    }

}

File: 프로그래머스/2/12936. 줄 서는 방법/줄 서는 방법.js
function solution(n, k) {
    
    var permu = 1;
    const people = [];
    
    for(let i=1; i<=n; i++){
        permu *= i;
        people.push(i);
    }
    
    var answer = [];
    
    while(people.length){
        if(people.length == 1){
            answer.push(people.pop());
            break;
        }
        
        permu /= people.length;
        var num = Math.floor((k-1)/permu);
        answer.push(people[num]);
        k -= num*permu;
        people.splice(num,1);
    }
    
    return answer;
}